sdk/search/azure-search-documents/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,64 @@
+# Troubleshooting Azure Search SDK Issues
+
+The `azure-search-documents` package provides APIs for operations on the 
+[Azure Cognitive Search](https://docs.microsoft.com/azure/search/search-what-is-azure-search) cloud service.
+
+## Table of Contents
+
+* [Identifying and Troubleshooting Issues by Response Code](#troubleshooting-issues-by-response-code)
+    * [HTTP 207 Errors](#207-multi-status)
+    * [HTTP 404 Errors](#404-not-found)
+    * [HTTP 429 Errors](#429-too-many-requests)
+* [Unexpected search query results](#unexpected-search-query-results)
+
+## Troubleshooting Issues By Response Code
+
+See [this page](https://docs.microsoft.com/rest/api/searchservice/http-status-codes) for the common response status 
+codes sent by the Azure Cognitive Search service.
+
+### 207 Multi-Status
+
+This response status indicates a partial success for an indexing operation. Some documents were successfully processed, 
+but at least one failed. Details of the failed documents are present in the individual `IndexingResult` objects within 
+the `IndexDocumentsResult`. If you want the [`indexDocumentsWithResponse`](https://docs.microsoft.com/java/api/com.azure.search.documents.searchclient.indexdocumentswithresponse?view=azure-java-stable) 
+method call to throw an exception on any failure, set [`IndexDocumentsOptions.throwOnAnyError`](https://docs.microsoft.com/java/api/com.azure.search.documents.models.indexdocumentsoptions.setthrowonanyerror?view=azure-java-stable#com-azure-search-documents-models-indexdocumentsoptions-setthrowonanyerror(boolean)) 
+to `true`. Each failure is then recorded in a separate `IndexingResult` and a single `IndexBatchException` is thrown by 
+the method.
+
+### 403 Forbidden
+
+Returned when you pass an invalid api-key. Search service uses two types of keys to control access: admin (read-write) 
+and query (read-only). The **admin key** grants full rights to all operations, including the ability to manage the service, 
+create and delete indexes, indexers, and data sources. The **query key** grants read-only access to indexes and documents. 
+Ensure that the key used for an API call provides sufficient privileges for the operation. See [here](https://docs.microsoft.com/azure/search/search-security-api-keys) 
+for details about managing API keys.
```

Comment:
Add link to Azure Identity troubleshooting guide for investigating other auth issues.